AnkiHubSoftware/ankihub_addon
# feat: Allow adding AnkiHub_Protect tags to multiple notes at once using dialog

This PR changes the dialog for protecting fields of notes using `AnkiHub_Protect` tags (see screenshot). (The dialog is reachable from the context menu when you select some notes in the Anki browser and then right click.)

**Changes**
- Previously it was only possible to change the tags for one note at once. Now the dialog can be used to do that for multiple notes.
- The text on the dialog is also changed, it now contains a link to the documentation about the `protected fields` feature.
- Previously the tags were updated when you closed the dialog, now it only updates tags when you click the `Ok` button on the dialog.
- Protecting fields using tags can now be undone (browser toolbar -> edit -> Undo [...])

<img src="https://github.com/ankipalace/ankihub_addon/assets/31575114/65bf140b-d446-4afc-82a8-1c6c2f32e930" width="400" />


## Related issues

Feature request on discourse: https://community.ankihub.net/t/allow-for-more-than-one-individually-note-field-protected/2331/3

## How to reproduce
Explain like if someone who doesn't know this project is reviewing your changes and how they can replicate this.

- Open the Anki browser.
- Select some AnkiHub notes.
- Right-click
- Choose `AnkiHub: Protect fields`
- Now you can select some of the fields in the dialog
- Click Ok
- A tooltip should be shown that says that the operation succeeded
- The notes should have the appropriate `AnkiHub_Protect` tags
